MapboxNavigationTests/NavigationViewControllerTests.swift

```diff
@@ -72,6 +72,21 @@ class NavigationViewControllerTests: XCTestCase {
         XCTAssertFalse(wayNameView.isHidden, "WayNameView should be visible.")
     }
     
+    var updatedStyleNumberOfTimes = 0
+    func testNavigationShouldNotCallStyleManagerDidRefreshAppearanceMoreThanOnce() {
+        let navigationViewController = NavigationViewController(for: initialRoute, styles: [DayStyle()])
```

Comment:
Nice catch.
Test using two styles, we guard against one style in `resetTimeOfDayTimer()` and the same should be true for `applyStyle`. There’s no reason why we should forcefully refresh appearance when only one style has been provided.

Reply:
Added test for two styles in https://github.com/mapbox/mapbox-navigation-ios/pull/1441/commits/b0d114f56d304b1f27d6f01bca0fdb1dd3bc3098